clangd-tidy.sh

```diff
@@ -0,0 +1,20 @@
+#! /bin/bash
+
+# Clangd by default will not run "expensive" checks. So we need two config
+# files: one for the language server; the other for the CI.
+# Because clangd-tidy does not support separate paths for config files,
+# we break these into separate files and do this moving hack.
+cp .clangd .clangd-orig
+echo "Diagnostics:" >> .clangd
+echo "  ClangTidy:" >> .clangd 
+echo "    FastCheckFilter: None" >> .clangd
```

Comment:
This will cause the clangd LSP to go haywire as it autoloads any changes to the file while running.

Reply:
I now provide two scripts: 

* `clangd-tidy.sh` that runs without changing the `.clangd` file 
* `clangd-tidy-ci.sh` that does update the `.clangd` file to use all checks. It still has the same logic around file renaming and movement in case we want to run this locally.